frame/contracts/src/tests.rs

```diff
@@ -2736,3 +2736,167 @@ fn get_runtime_storage() {
+								(call $ext_hash_twox_128
+									(local.get 3) ;; Input data from scratch buffer.
+									(local.get 4) ;; The length of the input buffer.
+									(local.get 2) ;; Write back into the result linear memory position.
+								)
+							)
+							(local.set 0 (i32.const 32)) ;; 256-bits output buffer size.
+							(call $ext_hash_twox_256
+								(local.get 3) ;; Input data from scratch buffer.
+								(local.get 4) ;; The length of the input buffer.
+								(local.get 2) ;; Write back into the result linear memory position.
+							)
+						)
+						(local.set 0 (i32.const 16)) ;; 128-bits output buffer size.
+						(call $ext_hash_blake2_128
+							(local.get 3) ;; Input data from scratch buffer.
+							(local.get 4) ;; The length of the input buffer.
+							(local.get 2) ;; Write back into the result linear memory position.
+						)
+					)
+					(local.set 0 (i32.const 32)) ;; 256-bits output buffer size.
+					(call $ext_hash_blake2_256
+						(local.get 3) ;; Input data from scratch buffer.
+						(local.get 4) ;; The length of the input buffer.
+						(local.get 2) ;; Write back into the result linear memory position.
+					)
+				)
+				(local.set 0 (i32.const 32)) ;; 256-bits output buffer size.
+				(call $ext_hash_keccak_256
+					(local.get 3) ;; Input data from scratch buffer.
+					(local.get 4) ;; The length of the input buffer.
+					(local.get 2) ;; Write back into the result linear memory position.
+				)
+			)
+			(local.set 0 (i32.const 32)) ;; 256-bits output buffer size.
+			(call $ext_hash_sha2_256
+				(local.get 3) ;; Input data from scratch buffer.
+				(local.get 4) ;; The length of the input buffer.
+				(local.get 2) ;; Write back into the result linear memory position.
+			)
+		)
+		;; Write back results of the hash function into the scratch buffer.
+		;; The test driver then asserts that the values are correct.
+		(call $ext_scratch_write
+			(local.get 2) ;; Linear memory location of the output buffer.
+			(local.get 0) ;; Number of output buffer bytes.
+		)
+	)
+)
+"#;
+
+#[test]
+fn crypto_hashes() {
+	let wasm_input = std::fs::read_to_string("src/hash_functions.wat")
+		.expect("couldn't open hash functions `.wat` test Wasm file");
+	let (wasm, code_hash) = compile_module::<Test>(&wasm_input).unwrap();
+
+	ExtBuilder::default().existential_deposit(50).build().execute_with(|| {
+		Balances::deposit_creating(&ALICE, 1_000_000);
+		assert_ok!(Contracts::put_code(Origin::signed(ALICE), 100_000, wasm));
+
+		// Instantiate the CRYPTO_HASHES contract.
+		assert_ok!(Contracts::instantiate(
+			Origin::signed(ALICE),
+			100_000,
+			100_000,
+			code_hash.into(),
+			vec![],
+		));
+		// Perform the call.
+		let input = b"0_DEAD_BEEF";
+		let result = <Module<Test>>::bare_call(
+			ALICE,
+			BOB,
+			0,
+			100_000,
+			input.to_vec(),
+		).unwrap();
+		assert_eq!(result.status, 0);
+		let expected = sp_io::hashing::blake2_256(input[1..].as_ref());
+		assert_eq!(&result.data, &expected)
```

Comment:
This currently does not work and `result.data` is always an empty vector. Need to investigate further.